src/components/Questionnaire/QuestionTypes/MedicationRequestQuestion.tsx

```diff
@@ -355,103 +356,113 @@ export function MedicationRequestQuestion({
                   "bg-transparent": !desktopLayout,
                 })}
               >
-                {medications.map((medication, index) => (
-                  <React.Fragment key={index}>
-                    {!desktopLayout ? (
-                      <Collapsible
-                        open={expandedMedicationIndex === index}
-                        onOpenChange={() => {
-                          setExpandedMedicationIndex(
-                            expandedMedicationIndex === index ? null : index,
-                          );
-                        }}
-                        className="border-b last:border-b-0"
-                      >
-                        <div
-                          className={cn(
-                            "flex items-center gap-2 px-2 py-0.5 rounded-md shadow-sm text-sm",
-                            expandedMedicationIndex === index
-                              ? "bg-gray-50"
-                              : "bg-gray-100",
-                          )}
+                {medications.map((medication, index) => {
+                  const isInactive = INACTIVE_MEDICATION_STATUSES.includes(
+                    medication.status as (typeof INACTIVE_MEDICATION_STATUSES)[number],
+                  );
+
+                  return (
+                    <React.Fragment key={index}>
```

Comment:
prefer using ids for keys over indexes, indexes tends to change when the array changes, here, use medication.id as key